admin_assignments_dates.php

```diff
@@ -1,517 +1,485 @@
+  <lable>Module 10</lable>
+     <input type="date" name="module_10_issued" id="module_10_issued" class="form-control" placeholder="Assignment 10" />
+	<br> 
+     <input type="date" name="module_10_deadl" id="module_10_deadl" class="form-control" placeholder="Assignment 10" />	  
+     
+    
+      </div>
+      <div class="modal-footer">
+                            	
+     <button type="submit" value="submit" id="update_id" name="updatedata" class="btn btn-info"> Update </button>
+   
+		  
+                         
+       <button type="button" class="btn btn-danger" data-dismiss="modal">Close</button>                  
+		 
+      </div>
+		</form>	
+    </div>
+  </div>
+</div>		
+		
+ 
+	
+		
+    </div>		  
+		 
+<table class="table table-bordered  "  style=" color: black; font-size: 12px; border:10px ">
+  <thead>
+    <tr>
+	 
+    <th scope="col">Assi ID</th>
+	<th scope="col">Batch ID</th>
+	<th scope="col"> M1 Issue </th>
+	<th scope="col"> M1 DL </th>
+	<th scope="col"> M2 Issue </th>
+	<th scope="col"> M2 DL </th>
+	<th scope="col"> M3 Issue </th>
+	<th scope="col"> M3 DL </th>
+    <th scope="col"> M4 Issue </th>
+	<th scope="col"> M4 DL </th>
+	<th scope="col"> M5 Issue </th>
+	<th scope="col"> M5 DL </th>
+     <th scope="col"> M6 Issue </th>
+	<th scope="col"> M6 DL </th>
+	<th scope="col"> M7 Issue </th>
+	<th scope="col"> M7 DL </th>
+     <th scope="col"> M8 Issue </th>
+	<th scope="col"> M8 DL </th>
+	<th scope="col"> M9 Issue </th>
+	<th scope="col"> M9 DL </th>
+	<th scope="col"> M10 Issue </th>
+	<th scope="col"> M10 DL </th>
+	<th scope="col" >Edit</th>
+	<th class="hidden" scope="col" >Delete</th>	
+		
+     
+    </tr>
+  </thead>
+  <tbody>
+	  <?php 
+ 		while ($row= $result->fetch_assoc()):  ?>  
+	  
+	<tr id="row_<?php echo $row['assignmnt_id']; ?>">
+		<td><?php echo $row['assignmnt_id']; ?></td>
+		<td><?php echo $row['batch_id']; ?></td>
+		<td><button type="button" class="btn btn-primary btn-xs" data-toggle="collapse" data-target="#assignmentDetails_<?php echo $row['assignmnt_id']; ?>"><i class="fa fa-plus"></i></button></td>
+		<td><button type="button" class="btn btn-success btn-xs editbtn"><i class="fa fa-cog" aria-hidden="true"></i></button></td>
+		<td><button type="button" class="btn btn-danger btn-xs deletebtn" data-id="<?php echo $row['assignmnt_id']; ?>"><i class="fa fa-trash"></i></button></td>
+	</tr>
+	<tr id="assignmentDetails_<?php echo $row['assignmnt_id']; ?>" class="collapse">
+		<td colspan="4">
+			<table class="table-condensed">
+				<thead>
+					<tr>
+						<th>Module Issued</th>
+						<th>Module Deadline</th>
+					</tr>
+				</thead>
+				<tbody>
+					<?php for ($i = 1; $i <= 10; $i++) { ?>
+						<?php if (!empty($row["module_{$i}_issued"]) && !empty($row["module_{$i}_deadl"])) { ?>
+							<tr>
+								<td><?php echo $row["module_{$i}_issued"]; ?></td>
+								<td><?php echo $row["module_{$i}_deadl"]; ?></td>
+							</tr>
+						<?php } ?>
+					<?php } ?>
+				</tbody>

```

Comment:
f the table has a lot of columns, it may not be practical to display all of them in one row. It is better to consider grouping some columns or hiding some information under buttons that can be expanded when needed.Additionally, there is an unnecessary class attribute "hidden" on the second column. It's also possible to use template literals for cleaner code and easier readability. 

In the optimized code, a button is added to reveal more details for each assignment. The details are displayed in a new row and only show the issued and deadline date for each module if they exist. The edit and delete buttons are simplified to icons, and we use data attributes for the delete button ID.